src/components/Global/TokenSelector/TokenSelector.tsx

```diff
@@ -94,8 +95,23 @@ const TokenSelector = ({ classNameButton, shouldBeConnected = true, onReset }: _
             balancesToDisplay = balances.filter((balance) => balance.chainId.toString() === safeInfo.chainId.toString())
         }
 
+        if (balancesToDisplay.length === 0) {
+            let tokens = selectedChainID
+                ? peanutTokenDetails.find((token) => token.chainId === selectedChainID)?.tokens
+                : peanutTokenDetails[0].tokens
+            tokens = tokens || peanutTokenDetails[0].tokens
+            balancesToDisplay = tokens.slice(0, 6).map((token: IToken) => ({
+                ...token,
+                chainId: selectedChainID || token.chainId,
+                price: 0,
+                amount: 0,
+                currency: '',
+                value: '',
+            }))
```

Comment:
nit: I think ideally we'd have a comment here in the code explaining why top 6 is the choice (and not top 200 or why we dont need to rank them)

Reply:
changed quite a bit, added comment on filters